### PR TITLE
Remove env concept from app config file

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
This PR disables updating the environment via the application config file.

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipe/issues/3019

**Does this PR introduce a user-facing change?**:

```release-note
Disable to update the environment via your application config file
```

### CODE REVIEW

**Comment by knanao** [comment]
Nice!
/lgtm

File: pkg/config/application.go [hunk 1]
```diff
@@ -26,20 +26,13 @@ const (
...
-	EnvLabelKey = "env"
 )
 
 type GenericApplicationSpec struct {
 	// The application name.
 	// This is required if you set the application through the application configuration file.
 	Name string `json:"name"`
-	// The environment name. You need to make sure that the environment name is unique in your project.
-	// Deprecated.
-	EnvName string `json:"envName"`
```
Comment (nghialv) [inline pkg/config/application.go:39]: How about just keeping this field to ensure that all kinds of config could be loadable?

Comment (nghialv) [reply pkg/config/application.go:39]: Because currently unknown fields are being disallowed while loading the config.

Comment (nakabonne) [reply pkg/config/application.go:39]: I would say `json.Unmarshal()` is able to parse one that has unknown field like `envName`, but am I correct?

Comment (nakabonne) [reply pkg/config/application.go:39]: https://go.dev/play/p/11XRE19yJv2

Comment (khanhtc1202) [reply pkg/config/application.go:39]: I thought as same as @nakabonne even without that field, json.Unmarshal() could work, right?

Comment (nghialv) [reply pkg/config/application.go:39]: https://github.com/pipe-cd/pipe/blob/master/pkg/config/config.go#L174

Comment (nghialv) [reply pkg/config/application.go:39]: We are having a customized one.

Comment (khanhtc1202) [reply pkg/config/application.go:39]: looks like it works 👍 
https://go.dev/play/p/X5pA17Zo0lQ

Comment (nakabonne) [reply pkg/config/application.go:39]: Oops!

**Comment by nakabonne** [comment]
Reverted!

**Comment by nghialv** [comment]
Thank you.
/approve

**Pull request merged**